niceml/utilities/ioutils.py

```diff
@@ -37,7 +37,8 @@ def list_dir(
     """
     cur_fs: AbstractFileSystem = file_system or LocalFileSystem()
     files: List[str] = [
-        relpath(cur_file, path) for cur_file in list(cur_fs.listdir(path, detail=False))
+        relpath(cur_file, path)
```

Comment:
I would like to retain the option to use the cache.

Reply:
```python
def list_dir(
    path: str,
    return_full_path: bool = False,
    recursive: bool = False,
    file_system: Optional[AbstractFileSystem] = None,
    filter_ext: Optional[List[str]] = None,
enable_cache:bool=False
) -> List[str]:
    """
    Returns a list of files in a directory. An AbstractFileSystem can be
    specified optionally. Otherwise, LocalFileSystem is used by default.

    Args:
        path: path to directory, which should be listed
        return_full_path: Returns full filepaths (True) or relative path (False)
        recursive: Determine if the function should look into subfolders
        file_system: Allow the function to be used with different file systems; default = local
        filter_ext: List of file extension to filter for; default = all files

    Returns:
        A list of files in the specified directory
    """
    cur_fs: AbstractFileSystem = file_system or LocalFileSystem()
    files: List[str] = [
        relpath(cur_file, path)
        for cur_file in list(cur_fs.ls(path, detail=False, refresh=True) if not enable_cache else cur_fs.list_dir(path, detail=False, refresh=True))
    ]
    ....
````

Reply:
Something like this.